### PR TITLE
feat: add tail mode for events and event deliveries

### PR DESCRIPTION
**Pull request by Oluwadaminiola**

### CODE REVIEW

File: web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts [hunk 1]
```diff
@@ -116,20 +116,20 @@ export class EventLogsComponent implements OnInit {
...
-		this.getEventLogs({ ...data, showLoader: true });
+		this.checkIfTailModeIsEnabled(data);
 	}
 
 	getSelectedDateRange(dateRange: { startDate: string; endDate: string }) {
 		this.eventsDateFilterFromURL = dateRange;
 		const data = this.addFilterToURL(dateRange);
 		clearInterval(this.getEventsInterval);
-		this.getEventLogs({ ...data, showLoader: true });
+		this.checkIfTailModeIsEnabled(data);
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts:132]: This `checkIfTailModeIsEnabled` shouldn't replace the `getEventLogs` here; it should simply return the status of tail mode, and then the required action should be taken here.

File: web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts [hunk 2]
```diff
@@ -181,21 +181,32 @@ export class EventLogsComponent implements OnInit {
...
+		if (this.enableTailMode) this.getEventLogs({ ...data, showLoader: true }).then(() => this.getEventsAtInterval({ ...data }));
+		else this.getEventLogs({ ...data, showLoader: true });
+	}
+
+	toggleTailMode(e: any) {
+		const tailModeConfig = e.target.checked;
+		this.enableTailMode = tailModeConfig;
+		localStorage.setItem('EVENT_LOGS_TAIL_MODE', JSON.stringify(tailModeConfig));
+		const data = this.getFiltersFromURL();
+		if (tailModeConfig) this.getEventLogs({ ...data, showLoader: true }).then(() => this.getEventsAtInterval({ ...data }));
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts:199]: This method should not be called here, this function should simply toggle tail mode and return the status to whoever called it.

**Review by horlah — CHANGES_REQUESTED** [review]

**Comment by horlah** [comment]
More issues:
- Wrong font size on the text
- The checkbox, cannot be unchecked

File: web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.html [hunk 3]
```diff
@@ -77,16 +77,17 @@
...
+			#tailModeToggle
 			type="checkbox"
 			class="peer w-24px h-18px border border-new.primary-50 rounded-4px checked:border-new.primary-400 checked:bg-new.primary-25 focus:bottom-2 focus:border-new.primary-400 checked:bg bg-white-100"
 			name="enableTailMode"
 			[checked]="enableTailMode"
 			id="tailMode"
 			(change)="toggleTailMode($event)"
 		/>
-		<img src="assets/img/check-icon-primary.svg" class="w-12px hidden peer-checked:block absolute top-[26%] right-[3px]" alt="check icon" />
+		<img src="assets/img/check-icon-primary.svg" class="w-12px hidden peer-checked:block absolute top-[20%] right-[3px]" alt="check icon" (click)="tailModeToggle.click()" />
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.html:90]: This shouldn't require a click here. Doesn't the checkbox support uncheck? @Oluwadaminiola

File: web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts [hunk 4]
```diff
@@ -92,7 +92,7 @@ export class EventLogsComponent implements OnInit {
 
 	async ngOnInit() {
 		const data = this.getFiltersFromURL();
-		this.checkIfTailModeIsEnabled(data);
+		this.checkIfTailModeIsEnabled() ? this.refetchEvents(data) : this.getEventLogs({ ...data, showLoader: true });
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts:95]: Do you really need to prefetch events?
You can just start the retry interval instead.

**Review by horlah — CHANGES_REQUESTED** [review]

File: web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts (hunk 4 above)
Comment (Oluwadaminiola) [reply web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts:95]: I added this for the loader effect, it's really just for user experience

Comment (horlah) [reply web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts:95]: You want to show the loader effect after enabling the tail mode?

File: web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts [hunk 5]
```diff
@@ -195,16 +195,23 @@ export class EventLogsComponent implements OnInit {
 		this.enableTailMode = tailModeConfig;
 		localStorage.setItem('EVENT_LOGS_TAIL_MODE', JSON.stringify(tailModeConfig));
 		const data = this.getFiltersFromURL();
-		if (tailModeConfig) this.getEventLogs({ ...data, showLoader: true }).then(() => this.getEventsAtInterval({ ...data }));
+		if (tailModeConfig) this.startTailing();
 		else {
 			clearInterval(this.getEventsInterval);
 			this.getEventLogs({ ...data, showLoader: true });
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/pages/project/event-logs/event-logs.component.ts:201]: We should also add a 3s wait here before calling this guy, so it is uniform.

If you'll have to write it in multiple places, you can as well make it a function.

**Review by horlah — CHANGES_REQUESTED** [review]

**Pull request merged**